Bini / Pett Script Modifications

nosrettep opened the pull request.

nosrettep committed:
Update script.js

nosrettep committed:
delete unused variables + comment out logging statements

nosrettep committed:
deleted more comments

nosrettep committed:
Update script.js

nosrettep commented:
Changed 3 things:

1.) Tenzin -> Bini.
2.) Add tracking for **Draws** (*note that it assumes all draws are 0RR MMR change and does not recognize non-zero RR games as draws*).
3.) Use `latestRawEloThisStream` and `earliestRawEloThisStream` to calculate the stream's total RR delta, which fixes an issue where the loss protection and promotion bonus RR were not being accounted for.

squarebracket commented on script.js:
```diff
@@ -28,23 +29,41 @@
+    
+    let latestMatchThisStream = 0;
+    let latestRawEloThisStream = null;
+    let earliestMatchThisStream = 100000000000000;
```
I assume you're using this for "date this is at an arbitrary point in the future". You can just use `Number.POSITIVE_INFINITY` here.

nosrettep replied:
```suggestion
    let earliestMatchThisStream = Number.POSITIVE_INFINITY;
```

nosrettep committed:
POSITIVE_INFINITY instead of 1000000000000

nosrettep committed:
use priorMatch to get proper original rawElo value

nosrettep committed:
Update script.js

nosrettep committed:
Update script.js

nosrettep committed:
const -> let

nosrettep committed:
Update script.js

nosrettep committed:
Update script.js

nosrettep committed:
revert changes back to where it wasnt erroring

there is still an issue wit hthe logic on line 60, which doesnt account for mmrChange being different than the true changed rr amount due to loss protection or promotion bonus RR

nosrettep committed:
add `playerName` parameter so it works for Luck

nosrettep committed:
POSITIVE_INFINITY
